tests/integration_tests/modules/test_package_update_upgrade_install.py

```diff
@@ -0,0 +1,76 @@
+"""Integration test for the package update upgrade install module.
+
+This test module asserts that packages are upgraded/updated during boot
+with the ``package_update_upgrade_install`` module. We are also testing
+if we can install new packages during boot too.
+
+(This is ported from
+``tests/cloud_tests/testcases/modules/package_update_upgrade_install.yaml``.)
+
+NOTE: the testcase for this looks for the command in history.log as
+      /usr/bin/apt-get..., which is not how it always appears. it should
+      instead look for just apt-get...
+NOTE: this testcase should not require 'apt_up_out', and should look for a
+      call to 'apt-get upgrade' or 'apt-get dist-upgrade' in cloud-init.log
+      rather than 'Calculating upgrade...' in output."""
+
+import re
+import pytest
+
+
+USER_DATA = """\
+#cloud-config
+packages:
+  - sl
+  - tree
+package_update: true
+package_upgrade: true
+"""
+
+
+@pytest.mark.user_data(USER_DATA)
+class TestPackageUpdateUpgradeInstall:
+
+    def assert_package_installed(self, pkg_out, name, version=None):
```

Comment:
@TheRealFalcon This assertion helper mirrors one in the `cloud_tests/` which is used in multiple places: do you have any thoughts as to whether we should store such shared "test" code on `client`, or elsewhere?

(No change needed as part of this PR, to be clear!)

Reply:
Yeah, that's a good question. I'm worried about seeing every conceivable helper function going into that class with implementations being nothing more than a series of execs or push/pulls and parsing what's returned, though I realize we're kind of already doing that with the `install_*` and read/write methods. 

Would it make sense to start putting these kinds of helper functions into separate files (not named util.py) and have the helper function take an `IntegrationClient` as an argument?